url86119f019a_solzy_hive_parse_tgz-pJ8-org_apache_hadoop_hive_ql_parse_MainJ8/src/org/apache/hadoop/hive/ql/parse/HiveParser_FromClauseParser.java

```diff
@@ -0,0 +1,6078 @@
+			case KW_STRING:
+			case KW_STRUCT:
+			case KW_TABLE:
+			case KW_TABLES:
+			case KW_TBLPROPERTIES:
+			case KW_TEMPORARY:
+			case KW_TERMINATED:
+			case KW_TEXTFILE:
+			case KW_TIMESTAMP:
+			case KW_TINYINT:
+			case KW_TO:
+			case KW_TOUCH:
+			case KW_TRIGGER:
+			case KW_TRUE:
+			case KW_TRUNCATE:
+			case KW_UNARCHIVE:
+			case KW_UNDO:
+			case KW_UNION:
+			case KW_UNIONTYPE:
+			case KW_UNLOCK:
+			case KW_UNSET:
+			case KW_UNSIGNED:
+			case KW_UPDATE:
+			case KW_USE:
+			case KW_USER:
+			case KW_USING:
+			case KW_UTC:
+			case KW_UTCTIMESTAMP:
+			case KW_VALUE_TYPE:
+			case KW_VIEW:
+			case KW_WHILE:
+			case KW_WITH:
+				{
+				alt19=2;
+				}
+				break;
+			case LPAREN:
+				{
+				alt19=3;
+				}
+				break;
+			default:
+				if (state.backtracking>0) {state.failed=true; return retval;}
+				NoViableAltException nvae =
+					new NoViableAltException("", 19, 0, input);
+				throw nvae;
+			}
+			switch (alt19) {
+				case 1 :
+					// FromClauseParser.g:146:6: ( Identifier LPAREN )=> partitionedTableFunction
+					{
+					pushFollow(FOLLOW_partitionedTableFunction_in_fromSource838);
+					partitionedTableFunction65=partitionedTableFunction();
+					state._fsp--;
+					if (state.failed) return retval;
+					if ( state.backtracking==0 ) adaptor.addChild(root_0, partitionedTableFunction65.getTree());
+
+					}
+					break;
+				case 2 :
+					// FromClauseParser.g:146:55: tableSource
+					{
+					pushFollow(FOLLOW_tableSource_in_fromSource842);
+					tableSource66=tableSource();
+					state._fsp--;
+					if (state.failed) return retval;
+					if ( state.backtracking==0 ) adaptor.addChild(root_0, tableSource66.getTree());
+
+					}
+					break;
+				case 3 :
+					// FromClauseParser.g:146:69: subQuerySource
+					{
+					pushFollow(FOLLOW_subQuerySource_in_fromSource846);
+					subQuerySource67=subQuerySource();
+					state._fsp--;
+					if (state.failed) return retval;
+					if ( state.backtracking==0 ) adaptor.addChild(root_0, subQuerySource67.getTree());
+
+					}
+					break;
+
+			}
+
+			// FromClauseParser.g:146:85: ( lateralView ^)*
+			loop20:
+			while (true) {
+				int alt20=2;
+				int LA20_0 = input.LA(1);
+				if ( (LA20_0==KW_LATERAL) ) {
+					alt20=1;
+				}
+
+				switch (alt20) {
+				case 1 :
+					// FromClauseParser.g:146:86: lateralView ^
+					{
+					pushFollow(FOLLOW_lateralView_in_fromSource850);
+					lateralView68=lateralView();
+					state._fsp--;
+					if (state.failed) return retval;
+					if ( state.backtracking==0 ) root_0 = (CommonTree)adaptor.becomeRoot(lateralView68.getTree(), root_0);
+					}
+					break;
+
+				default :
+					break loop20;
+				}
+			}
+
+			}
+
+			retval.stop = input.LT(-1);
+
+			if ( state.backtracking==0 ) {
+			retval.tree = (CommonTree)adaptor.rulePostProcessing(root_0);
+			adaptor.setTokenBoundaries(retval.tree, retval.start, retval.stop);
+			}
+			if ( state.backtracking==0 ) { gParent.msgs.pop(); }
+		}
+
+		catch (RecognitionException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: An exception is caught and re-thrown within the catch clause without any other operations being performed on the exception, making the catch clause unnecessary.

**Fix**: Perform additional operations on the exception that has been caught. You can wrap a standard exception into a custom exception to add additional information or special handling. When you do that, make sure to set the original exception as the cause. Otherwise, you will lose the stack trace and message of the original exception, which will make it difficult to analyze the event that caused the exception.  
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)